src/main/java/de/tum/p2p/api/netty/forwarding/OnionTunnelBuildMessage.java

```diff
@@ -0,0 +1,72 @@
+package de.tum.p2p.api.netty.forwarding;
+
+import de.tum.p2p.Peer;
+import de.tum.p2p.proto.message.MessageType;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.spec.X509EncodedKeySpec;
+
+import static de.tum.p2p.util.ByteBuffers.randPadRemaining;
+
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class OnionTunnelBuildMessage extends OnionApiMessage {
+
+    private Peer destinationPeer = null;
+
+    public OnionTunnelBuildMessage(Peer destinationPeer) {
+        super(MessageType.ONION_TUNNEL_BUILD);
+        this.destinationPeer = destinationPeer;
+    }
+
+    public static OnionTunnelBuildMessage me(Peer destinationPeer) { return new OnionTunnelBuildMessage(destinationPeer); }
+
+    public Peer getDestinationPeer() {
+        return destinationPeer;
+    }
+
+    @Override
+    protected ByteBuffer writeMessage(ByteBuffer typedMessageBuffer) {
+        return randPadRemaining(typedMessageBuffer); // Onion Cover message carries no payload
```

Comment:
If you dont expect this to be used/useful throw new UnsupportedOperationException() here